sdks/python/apache_beam/transforms/ptransform.py

```diff
@@ -118,14 +127,17 @@ def _allocate_materialized_result(pipeline):
 
 def _get_materialized_result(pipeline_id, result_id):
   with _pipeline_materialization_lock:
+    if pipeline_id not in _pipeline_materialization_cache:
+      raise Exception(
+          'Materialization in out-of-process and remote runners is not yet '
```

Comment:
Thanks, though I think we still need to handle the case of fork() is being used to spawn workers (maybe check PID as well as presence in dict?). 

Is this global rather than, say, methods on pipeline itself, just to support passing through pickling?